app-sync.test.js

```diff
@@ -0,0 +1,87 @@
+const fs = require('fs');
+const request = require('supertest');
+const { handlers: app } = require('./app-sync');
+const { getAccountDb } = require('./account-db');
+const { getPathForUserFile } = require('./util/paths');
+
+describe('/download-user-file', () => {
+  describe('default version', () => {
+    it('returns 401 if the user is not authenticated', async () => {
+      const res = await request(app).get('/download-user-file');
+
+      expect(res.statusCode).toEqual(401);
+      expect(res.body).toEqual({
+        details: 'token-not-found',
+        reason: 'unauthorized',
+        status: 'error'
+      });
+    });
+
+    it('returns 401 if the user is invalid', async () => {
+      const res = await request(app)
+        .get('/download-user-file')
+        .set('x-actual-token', 'invalid-token');
+
+      expect(res.statusCode).toEqual(401);
+      expect(res.body).toEqual({
+        details: 'token-not-found',
+        reason: 'unauthorized',
+        status: 'error'
+      });
+    });
+
+    it('returns 400 error if the file does not exist in the database', async () => {
+      const res = await request(app)
+        .get('/download-user-file')
+        .set('x-actual-token', 'valid-token')
+        .set('x-actual-file-id', 'non-existing-file-id');
+
+      expect(res.statusCode).toEqual(400);
+    });
+
+    it('returns 500 error if the file does not exist on the filesystem', async () => {
+      getAccountDb().mutate(
+        'INSERT INTO files (id, deleted) VALUES (?, FALSE)',
+        ['missing-fs-file']
+      );
+
+      const res = await request(app)
+        .get('/download-user-file')
+        .set('x-actual-token', 'valid-token')
+        .set('x-actual-file-id', 'missing-fs-file');
+
+      expect(res.statusCode).toEqual(500);
+    });
+
+    it('returns an attachment file', async () => {
+      fs.writeFileSync(getPathForUserFile('file-id'), 'content');
+      getAccountDb().mutate(
+        'INSERT INTO files (id, deleted) VALUES (?, FALSE)',
+        ['file-id']
+      );
+
+      const res = await request(app)
+        .get('/download-user-file')
+        .set('x-actual-token', 'valid-token')
+        .set('x-actual-file-id', 'file-id');
+
+      expect(res.statusCode).toEqual(200);
+      expect(res.headers).toEqual(
+        expect.objectContaining({
+          'content-disposition': 'attachment;filename=file-id',
+          'content-type': 'application/octet-stream'
+        })
+      );
+    });
+  });
+
+  // TODO: build the synced-up database support and enable the test cases
+  // Versioning example: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/locking-api-versions.html
+  describe('TBD-date (2022-11-25) version', () => {
```

Comment:
What is this for?

Reply:
Originally the idea was to allow customers to get the full synced up database (currently we have the initial database + a lot of CRDT actions, but no way to get everything in a single file from the server).

ButI don't think I'll actually do this anymore, so I removed the todo tests: https://github.com/actualbudget/actual-server/pull/91/commits/6b8b6584f1f458a768851520157ae90ac3641a14

Good catch!